### PR TITLE
Add films_with_biggest_crew, films_in_each_language, most_popular_directors, directors_with_biggest_amount_of_genres records

### PR DESCRIPTION
**Pull request by DanyloSarvasNulp**

### CODE REVIEW

**Review by velychkovych — CHANGES_REQUESTED** [review]
Wait until auto import branch will be merged, merge main into your branch and update this PR.

File: reports.py [hunk 1]
```diff
@@ -304,3 +304,115 @@ def directors_best_titles(context: dict[str, DataFrame]) -> DataFrame:
...
+        title_basics_transformed.alias("tb")
+        .filter(f.col("tb.titleType").isin("tvMovie", "movie"))
+        .join(crew_transformed.alias("c"), f.col("tb.tconst") == f.col("c.tconst"))
+        .join(name_basics.alias("nb"), f.col("c.director") == f.col("nb.nconst"))
+        .join(ratings.alias("r"), f.col("tb.tconst") == f.col("r.tconst"))
+        .groupBy("nb.nconst", "nb.primaryName")
+        .agg(f.countDistinct("tb.genre").alias("NumberOfGenres"))
+        .orderBy(f.desc("NumberOfGenres"))
+        .select(
+            f.col("nb.primaryName"),
```
Comment (velychkovych) [inline reports.py:409]: Here the same as comment above.

File: reports.py [hunk 2]
```diff
@@ -304,3 +304,115 @@ def directors_best_titles(context: dict[str, DataFrame]) -> DataFrame:
...
+        title_basics.alias("tb")
+        .filter(f.col("tb.titleType").isin("tvMovie", "movie"))
+        .join(crew_transformed.alias("c"), f.col("tb.tconst") == f.col("c.tconst"))
+        .join(name_basics.alias("nb"), f.col("c.director") == f.col("nb.nconst"))
+        .join(ratings.alias("r"), f.col("tb.tconst") == f.col("r.tconst"))
+        .groupBy("nb.nconst", "nb.primaryName")
+        .agg(f.sum("r.numVotes").alias("NumberOfReviews"))
+        .orderBy(f.desc("NumberOfReviews"))
+        .select(
+            f.col("nb.primaryName"),
```
Comment (velychkovych) [inline reports.py]: Here the same as comment above.

File: reports.py [hunk 3]
```diff
@@ -304,3 +304,115 @@ def directors_best_titles(context: dict[str, DataFrame]) -> DataFrame:
...
+    akas = context["akas"]
+
+    df = (
+        title_basics.alias("tb")
+        .join(akas.alias("a"), f.col("tb.tconst") == f.col("a.titleId"))
+        .groupBy("a.language")
+        .agg(f.count("a.titleId").alias("NumberOfFilms"))
+        .orderBy(f.desc("NumberOfFilms"))
+        .select(
+            f.col("a.language"),
```
Comment (velychkovych) [inline reports.py]: I would add an alias to this column, but it's up to you.

**Review by velychkovych — CHANGES_REQUESTED** [review]

**Review by C0l1ns — APPROVED** [review]

**Pull request merged**